### ISSUE TITLE
TypeError: unsupported operand type(s) for +: 'float' and 'decimal.Decimal'

### ISSUE THREAD

**Issue by Petru-Tanas-ProcessITCS** [jsvine/pdfplumber#290]
I am trying to extract a table from a random PDF file using explicit line, with the following table setting:

```python
table_settings={
    "vertical_strategy": "explicit", 
    "horizontal_strategy": "explicit",
    "explicit_vertical_lines": (324, 444),
    "explicit_horizontal_lines": (333, 444),
}
im.debug_tablefinder(table_settings)
```

traceback:
```python
TypeError                                 Traceback (most recent call last)
<ipython-input-33-6cb1c62ac2cf> in <module>
      5     "explicit_horizontal_lines": (333, 444),
      6 }
----> 7 im.debug_tablefinder(table_settings)

c:\costum\programs\python\lib\site-packages\pdfplumber\display.py in debug_tablefinder(self, tf)
    240             self.debug_table(table)
    241 
--> 242         self.draw_lines(tf.edges, stroke_width=1)
    243 
    244         self.draw_circles(

c:\costum\programs\python\lib\site-packages\pdfplumber\display.py in draw_lines(self, list_of_lines, **kwargs)
    128     def draw_lines(self, list_of_lines, **kwargs):
    129         for x in utils.to_list(list_of_lines):
--> 130             self.draw_line(x, **kwargs)
    131         return self
    132 

c:\costum\programs\python\lib\site-packages\pdfplumber\display.py in draw_line(self, points_or_obj, stroke, stroke_width)
    122             points = ((obj["x0"], obj["top"]), (obj["x1"], obj["bottom"]))
    123         self.draw.line(
--> 124             list(map(self._reproject, points)), fill=stroke, width=stroke_width
    125         )
    126         return self

c:\costum\programs\python\lib\site-packages\pdfplumber\display.py in _reproject(self, coord)
     99         rx0, rtop = self.root.bbox[:2]
    100         _x0 = (x0 + rx0 - px0) * self.scale
--> 101         _top = (top + rtop - ptop) * self.scale
    102         return (_x0, _top)
    103 

TypeError: unsupported operand type(s) for +: 'float' and 'decimal.Decimal'
```

python 3.8
windows 10
pdfplumber 0.5.23

**Comment by samkit-jain** [comment]
Hi @Petru-Tanas-ProcessITCS. Appreciate your interest in the library. Please use the following table settings.

```python
from decimal import Decimal

table_settings={
    "vertical_strategy": "explicit", 
    "horizontal_strategy": "explicit",
    "explicit_vertical_lines": (Decimal(324), Decimal(444)),
    "explicit_horizontal_lines": (Decimal(333), Decimal(444)),
}

im.debug_tablefinder(table_settings)
```

**Comment by jsvine** [comment]
Thanks, @samkit-jain — that's the correct approach for the current codebase. I think, though, that it probably makes sense to auto-`decimalize` the values passed for those settings, so I'm reopening this issue and adding the `enhancement` label.

**Comment by jsvine** [comment]
Added this enhancement, which is now available in v0.5.24. Thanks!
